src/lib/y2partitioner/actions/create_partition_table.rb

```diff
@@ -96,10 +94,95 @@ def run?
         false
       end
 
-      # Return the device name of the disk
-      # @return [String]
-      def disk_name
-        controller.disk_name
+      # If the operation have destructive consequences, ask for confirmation
+      # to the user.
+      #
+      # @return [Boolean] true if the action is safe or the user confirmed
+      #   to accept the consequences
+      def confirm
+        return true if safe?
+
+        if disk.filesystem
+          confirm_with_filesystem
+        else
+          confirm_with_nested_devices
+        end
+      end
+
+      # Whether performing the action can be considered non-destructive
+      #
+      # @return [Boolean] true if the disk is completely empty and unused or if
+      #   it only contains an empty partition table
+      def safe?
+        desc = disk.descendants.size
+        return true if desc.zero?
+        return false if desc > 1
+
+        disk.partition_table? || disk.encrypted?
+      end
+
+      # @see #confirm
+      def confirm_with_nested_devices
+        confirm_recursive_delete(
+          disk,
+          _("Confirm Deleting of Current Devices"),
+          _("If you proceed, the following devices will be deleted:"),
+          # TRANSLATORS %s is the disk device name ("/dev/sda" or similar)
+          _("Really create a new partition table on %s") % disk_name
+        )
+      end
+
+      # @see ConfirmRecursiveDelete#recursive_delete_yes_label
+      def recursive_delete_yes_label
+        Yast::Label.YesButton
+      end
+
+      # @see ConfirmRecursiveDelete#recursive_delete_no_label
+      def recursive_delete_no_label
+        Yast::Label.NoButton
+      end
+
+      # @see #confirm
+      def confirm_with_filesystem
+        fs = disk.filesystem.type.to_human
+        # TRANSLATORS: the first %s is replaced by the type of the filesystem
+        # (for example, "Btrfs" or "Ext2"), the second one by the disk device name
+        # ("/dev/sda" or similar).
+        msg = format(
+          _(
+            "This will delete the %s file system from the device\n\n" \
```

Comment:
maybe mention that it destroy all data on this partition. as noone care about fs, but data is what matters.